File: ThirtyFour.UI/ThirtyFour.UI/Form/Checkbox.cs
using OpenQA.Selenium;

namespace ThirtyFour.UI.Form
{
    public class Checkbox : BooleanElement
    {
        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="element">The checkbox element to wrap</param>
        public Checkbox(IWebElement element) : base(element) { }

        /// <summary>
        /// Sets the checked state of the element
        /// </summary>
        /// <param name="onOrOff">the boolean state of the element</param>
        public void SetChecked(bool onOrOff)
        {
            if ((onOrOff && !this.IsChecked) || (!onOrOff && this.IsChecked))
            {
                this.Element.Click();
            }
        }

    }
}

File: java/src/main/java/com/github/danhumphrey/thirtyfour/ui/form/Checkbox.java
package com.github.danhumphrey.thirtyfour.ui.form;

import org.openqa.selenium.WebElement;

/**
 * The Checkbox class provides helper methods for interacting with a HTML checkbox
 */
public class Checkbox extends BooleanElement {

	public Checkbox(WebElement element) {
		super(element);
	}

	/**
	 * Sets the checked state of the element
	 * @param onOrOff the boolean state of the element
	 */
	public void setChecked(boolean onOrOff) {
		if( (onOrOff && !getChecked()) || (!onOrOff && getChecked()) ) {
	        click();
	    }
	}
	
}
